modules/_labs/select/react/lib/SelectMenu.tsx

```diff
@@ -134,21 +196,131 @@ const MenuList = styled('ul')<Pick<SelectProps, 'error'>>(
   })
 );
 
-const SelectMenu = (props: SelectMenuProps) => {
-  const {children, error, isAnimated, isHiding, menuRef, ...elemProps} = props;
+export default class SelectMenu extends React.Component<SelectMenuProps, SelectMenuState> {
+  static defaultProps = {
+    isAnimated: true,
+    isAutoFlipped: true,
+    isAutoFocused: true,
+    isFlipped: false,
+    isHidden: false,
+    isHiding: false,
+    width: buttonDefaultWidth,
+  };
 
-  return (
-    <Menu error={error} isAnimated={isAnimated} isHiding={isHiding}>
-      <MenuList error={error} ref={menuRef} role="listbox" tabIndex={-1} {...elemProps}>
-        {children}
-      </MenuList>
-    </Menu>
-  );
-};
+  state: Readonly<SelectMenuState> = {
+    width: 0,
+  };
 
-SelectMenu.defaultProps = {
-  isAnimated: true,
-  isHiding: false,
-};
+  private generatePopperOptions = () => {
+    const {isAutoFlipped, isAutoFocused, isFlipped, menuRef} = this.props;
+
+    let fallbackPlacements: Placement[] = [];
+    if (isAutoFlipped) {
+      fallbackPlacements = isFlipped ? ['bottom-start'] : ['top-start'];
+    }
+
+    const modifiers = [
+      {
+        name: 'flip',
+        options: {
+          fallbackPlacements,
+        },
+      },
+      {
+        name: 'offset',
+        options: {
+          offset: ({
+            placement,
+            reference,
+            popper,
+          }: {
+            placement: Placement;
+            reference: Rect;
+            popper: Rect;
+          }) => {
+            // Skid menu along the edge of the button to account
+            // for fractional x-positioning of the button (e.g.,
+            // if the button is in a horizontally-centered modal)
+            // and to ensure proper alignment between the button
+            // and the menu
+            const skidding = reference.x % 1 === 0 ? 0 : 1;
+            // Displace menu towards the button to obscure the bottom
+            // edge of the button and to create a smooth visual
+            // connection between the button and the menu
+            const distance = -parseInt(borderRadius.m, 10);
```

Comment:
Clever hack here. `parseInt` will convert a string to a number from left-to-right until it encounters a non-number and will stop parsing and return the result so far. This effectively strips out the `px` part of the value.